### PR TITLE
 NEW: Introduce `desktop_only` arg into `fdl` and `node_defs`

### PR DESCRIPTION
**Pull request by ZaiZu2**
Dependency to:
- https://github.com/ForloopAI/forloop_platform/pull/899

### CODE REVIEW

File: forloop_modules/function_handlers/auxilliary/form_dict_list.py [hunk 1]
```diff
@@ -146,10 +156,10 @@ def button_image(self, image, x_size, y_size, x_offset, function, function_args=
         """
 
         key = "ButtonImage"
-        value = {"image": image, "function": function, "function_args": function_args, "x_size": x_size, "y_size": y_size, "x_offset": x_offset}
+        value = {"image": image, "function": function, "function_args": function_args, "x_size": x_size, "y_size": y_size, "x_offset": x_offset, 'desktop_only': desktop_only}
         self.insert_element_to_form_dict_list(key, value, row)
 
-    def checkbox(self, name, bool_value: bool = False, row=None):
+    def checkbox(self, name, bool_value: bool = False, row=None, desktop_only=False):
```
Comment (DovaX) [inline forloop_modules/function_handlers/auxilliary/form_dict_list.py:162]: row should be the last parameter

File: forloop_modules/function_handlers/auxilliary/form_dict_list.py [hunk 2]
```diff
@@ -128,10 +138,10 @@ def button(self, function, function_args, text, focused: bool = False, enforce_r
             row (int): row number of the element - specified only if it's not first element on the same row
         """    
         key = "Button"
-        value = {"name": name, "function": function, "function_args": function_args, "text": text, "focused": focused, "enforce_required": enforce_required, "frontend_implementation": frontend_implementation}
+        value = {"name": name, "function": function, "function_args": function_args, "text": text, "focused": focused, "enforce_required": enforce_required, "frontend_implementation": frontend_implementation, 'desktop_only': desktop_only}
```
Comment (DovaX) [inline forloop_modules/function_handlers/auxilliary/form_dict_list.py:141]: frontend_implementation + desktop_only are duplicates

File: forloop_modules/function_handlers/auxilliary/form_dict_list.py [hunk 3]
```diff
@@ -128,10 +138,10 @@ def button(self, function, function_args, text, focused: bool = False, enforce_r
...
         self.insert_element_to_form_dict_list(key, value, row)
 
-    def button_image(self, image, x_size, y_size, x_offset, function, function_args=None, row=None):
+    def button_image(self, image, x_size, y_size, x_offset, function, function_args=None, row=None, desktop_only=False):
```
Comment (DovaX) [inline forloop_modules/function_handlers/auxilliary/form_dict_list.py:144]: row should be the last parameter

File: forloop_modules/function_handlers/auxilliary/form_dict_list.py [hunk 4]
```diff
@@ -98,10 +108,10 @@ def combobox(self, name, options, multiselect_indices=None, default=None, show_i
             row (int, optional): Row number of the element - specified only if it's not first element on the same row. Defaults to None.
         """
         key = "Combobox"
-        value = {"name": name, "options": options, "default": default, "multiselect_indices": multiselect_indices, "show_info":show_info}
+        value = {"name": name, "options": options, "default": default, "multiselect_indices": multiselect_indices, "show_info":show_info, 'desktop_only': desktop_only}
         self.insert_element_to_form_dict_list(key, value, row)
 
-    def comboentry(self, name, text, options, show_info=None, row=None):
+    def comboentry(self, name, text, options, show_info=None, row=None, desktop_only=False):
```
Comment (DovaX) [inline forloop_modules/function_handlers/auxilliary/form_dict_list.py:114]: row should be the last parameter

File: forloop_modules/function_handlers/auxilliary/form_dict_list.py [hunk 5]
```diff
@@ -78,15 +78,25 @@ def entry(self, name, text, category=None, input_types=None, required=None, type
...
+        }
 
         if required:
             if "Label" in self[-1].keys():
                 self[-1]["Label"] = self[-1]["Label"] + "*"
 
         self.insert_element_to_form_dict_list(key, value, row)
 
-    def combobox(self, name, options, multiselect_indices=None, default=None, show_info=None, row=None):
+    def combobox(self, name, options, multiselect_indices=None, default=None, show_info=None, row=None, desktop_only=False):
```
Comment (DovaX) [inline forloop_modules/function_handlers/auxilliary/form_dict_list.py:99]: row should be the last parameter

File: forloop_modules/function_handlers/auxilliary/form_dict_list.py [hunk 6]
```diff
@@ -62,7 +62,7 @@ def label(self, text, row=None):
         value = text
         self.insert_element_to_form_dict_list(key, value, row)
 
-    def entry(self, name, text, category=None, input_types=None, required=None, type='text', file_types=None, show_info=None, row=None):
+    def entry(self, name, text, category=None, input_types=None, required=None, type='text', file_types=None, show_info=None, row=None, desktop_only=False):
```
Comment (DovaX) [inline forloop_modules/function_handlers/auxilliary/form_dict_list.py:65]: row should be the last parameter

**Review by DovaX — APPROVED** [review]

File: forloop_modules/function_handlers/auxilliary/form_dict_list.py (hunk 2 above)
Comment (ZaiZu2) [reply forloop_modules/function_handlers/auxilliary/form_dict_list.py:141]: I reached out to Tomas and Joaquin for clarification - I believe `frontend_implementation` stores the information if the button should be implemented in webapp with custom logic. `desktop_only` stores the information if the buttons should be rendered in webapp.

**Pull request merged**